driver/adam_driver.hpp

```diff
@@ -225,8 +235,8 @@ class AdamDriver : public Driver
     miopenDataType_t grad_type;
 };
 
-template <typename Tgpu, typename Tref, bool is_amp, typename Tgrad>
-int AdamDriver<Tgpu, Tref, is_amp, Tgrad>::ParseCmdLineArgs(int argc, char* argv[])
+template <typename Tgpu, typename Tref, bool adamw, bool is_amp, typename Tgrad>
+int AdamDriver<Tgpu, Tref, adamw, is_amp, Tgrad>::ParseCmdLineArgs(int argc, char* argv[])
```

Comment:
I missed a previous PR, but since a variable has been added here, I would ask is it necessary having all those variables as a template parameters?
At least `is_amp` and `adamw` should be runtime parameters to avoid code bloating and long compilation time.